sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -523,6 +523,23 @@ int64_t CurlSession::Read(Azure::Core::Context const& context, uint8_t* buffer,
   totalRead = ReadFromSocket(buffer, static_cast<size_t>(readRequestLength));
   this->m_sessionTotalRead += totalRead;
 
+  // Reading 0 bytes means closed connection.
+  // For known content length and chunked response, this means there is nothing else to read from
+  // server or lost connection before getting full response.
+  // For unknown response size, it means the end of response and it's fine.
+  if (totalRead == 0 && (this->m_contentLength > 0 || this->m_isChunkedResponseType))
```

Comment:
Nice comment. Add tests?

Reply:
Tests are hard for this one.
I need to simulate a connection lost in the middle of a read operation or find a Rest Server that would return a shorter response than what it specifies in content-length header.

What other languages did to simulate the read errors for a stream is to add a hook in the code that we can use to force a read error behavior. I added at some point before but it was not welcome by the team (and I honestly don't like it as well).

So, may be after having playback/recording test framework enabled, I can just manually create a bad response to test this.